examples/homotopy/plot_homotopy.py

```diff
@@ -0,0 +1,106 @@
+import json
+from pathlib import Path
+
+import copt
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy.io as sio
+from copt.constraint import (ElementWiseInequalityConstraint,
+                             RowEqualityConstraint)
+from copt.homotopy import minimize_homotopy_cgm
+from copt.loss import LinearObjective
+
+basepath = Path(".")
+def reduced_digits():
+    mat = sio.loadmat(basepath / "data/reduced_clustering_mnist_digits.mat")
```

Comment:
It would be nice if this dataset was generally available from the library (https://github.com/openopt/copt/blob/master/copt/datasets.py) and not just specific for this example. This way, we could re-use it easier from other examples